independent-projects/bootstrap/maven-resolver/src/main/java/io/quarkus/bootstrap/resolver/maven/workspace/WorkspaceLoader.java

```diff
@@ -272,6 +272,13 @@ public Model resolveEffectiveModel(String groupId, String artifactId, String ver
                 : null;
     }
 
+    private Path resolveProjectPath(Path pomFile) {
+        if (pomFile.endsWith("target/.flattened-pom.xml")) {
```

Comment:
I don't like this strategy much (hence why this PR is in draft). Any suggestions?